src/clj/clojurians_log/db/queries.clj

```diff
@@ -23,8 +23,28 @@
 (defn- assoc-inst [message]
   (assoc message :message/inst (time-util/ts->inst (:message/ts message))))
 
+(defn channel-thread-messages-of-day
+  "Retrieve all messages for threads that started in the given channel on the given day"
+  [db chan-name day]
+  (->> (d/q '[:find [(pull ?msg [:message/text :message/ts :message/thread-ts {:message/user [:user/name :user-profile/image-48]}]) ...]
```

Comment:
It's not very nice that the pull pattern for a single message is duplicated here (also present in `channel-day-messages`. Ideally, it the pattern would be defined in another var. Had a bit of trouble because the query pattern is a quoted vector.

Pointers would be appreciated.

Reply:
ping @plexus. Not entirely sure if you'd be notified otherwise.

Reply:
In general when you need to build up queries programmatically the advice seems to be to use map notation, rather than vectors ([see datomic docs](https://docs.datomic.com/on-prem/query.html#building-queries-programmatically))

So maybe something like this?

``` clojure
(def pull-message-pattern
  '[(pull ?msg
          [:message/text
           :message/ts
           :message/thread-ts
           {:message/user [:user/name
                           :user-profile/image-48]}])
    ...])

(defn channel-thread-messages-of-day
  "Retrieve all messages for threads that started in the given channel on the given day"
  [db chan-name day]
  (->> (d/q {:find [pull-message-pattern]
             :in '[$ ?chan-name ?day [?from-date ?to-date]]
             :where '[[?msg  :message/channel ?chan]
                      [?chan :channel/name ?chan-name]
                      [?msg  :message/thread-inst ?thread-inst]
                      [(.after ^java.util.Date ?thread-inst ?from-date)]
                      [(.before ^java.util.Date ?thread-inst ?to-date)]]}
            db
            chan-name
            day
            (time-util/day-str->date-interval day))
       (map assoc-inst)
       (sort-by :message/inst)))
```